Utils/Dataflow/test/utils/reconstruction.py

```diff
@@ -0,0 +1,99 @@
+"""
+Task chain reconstruction from elasticsearch.
+"""
+import copy
+import sys
+
+import elasticsearch
+
+
+INDEX = 'tasks_chain_data'
+
+
+def es_connect():
+    ''' Establish a connection to elasticsearch.
+
+    TODO: take parameters from es config.
+    '''
+    return elasticsearch.Elasticsearch(['localhost:9200'])
+
+
+def get_chain_id(es, index, taskid):
+    ''' Get chain_id by given taskid. '''
+    srch = {'query': {'match': {'_id': taskid}}}
+    results = es.search(index=index, body=srch)
+    results = results['hits']['hits']
+    if not results:
+        return False
+    else:
+        return results[0]['_source'].get('chain_id')
+
+
+def get_chain_data(es, index, chain_id):
+    ''' Get chain_data of all tasks that have given chain_id. '''
+    srch = {'query': {'match': {'chain_id': chain_id}}}
+    rtrn = []
+    fr = 0
+    sz = 100
+    while True:
+        results = es.search(index=index, body=srch, from_=fr, size=sz)
```

Comment:
It is up to Mikhail, but I think if we\`re going to keep all the data in memory (`rtrn` variable) anyway, then it is OK to get them all at one gulp.
I mean of course we don\`t know how long the chain is, but still it must be OK to set `sz` to something like 2000 (used by Mikhail [here](https://github.com/PanDAWMS/panda-bigmon-atlas/blob/devel/atlas/dkb/views.py#L145) and [here](https://github.com/PanDAWMS/panda-bigmon-atlas/blob/devel/atlas/dkb/views.py#L643)).

Reply:
Done.